notoptions-mitigation.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Plugin Name: VIP NotOptions Mitigation
+ * Description: Detect invalid values in the notoptions cache.
+ * Author: Automattic
+ * License: GPL version 2 or later - http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+ */
+
+namespace Automattic\VIP;
+
+use Automattic\VIP\Utils\Alerts;
+
+if ( is_admin() || ( defined( 'WP_CLI' ) && WP_CLI ) ) {
+	add_action( 'muplugins_loaded', __NAMESPACE__ . '\notoptions_mitigation', -999 );
+}
+
+function notoptions_mitigation() {
+	global $wpdb;
+	$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+	if ( ! is_array( $notoptions ) ) {
+		return;
+	}
+
+	// filter for any values not equal to (bool)true.
+	$not_trues = array_filter( $notoptions, function( $v ) {
+		return true !== $v;
+	} );
+
+	$total_invalid = count( $not_trues );
+
+	// if they exist, something's borked
+	if ( 1 <= $total_invalid ) {
+
+		// attempt repair
+		$flushed = wp_cache_delete( 'notoptions', 'options' );
+
+		// begin prepping notification
+		$is_vip_env    = ( defined( 'WPCOM_IS_VIP_ENV' ) && true === WPCOM_IS_VIP_ENV );
+		$environment   = ( ( defined( 'VIP_GO_APP_ENVIRONMENT' ) && VIP_GO_APP_ENVIRONMENT ) ? VIP_GO_APP_ENVIRONMENT : 'unknown' );
+		$site_id       = defined( 'VIP_GO_APP_ID' ) ? VIP_GO_APP_ID : false;
+
+		// Send notices to VIP staff if this is happening on VIP-hosted sites
+		if ( $is_vip_env && $site_id ) {
+
+			$irc_alert_level = 2; // ALERT
+			$opsgenie_alert_level = 'P4';
+
+			$subject = 'NOTOPTIONS: %s (%s VIP Go site ID: %s%s) %s invalid values found in notoptions. Cache flush was %ssuccessful.';
```

Comment:
If it's a multi-site, it might be helpful to know which subsite this occurred on.